hedyorg/hedy
# [TESTING] Typescript instrumentation for code coverage

**Description**

Fixes typescript implementation in code coverage. The trick, @jpelay, was to add the instrumentation to the TypeScript code before compiling it. I also only compile with esbuild, but I believe you can interject `tsc` as you had it before, as long as you are running `tsc` on the instrumented TypeScript. I would test all of this but I ran out of time :)

I fear this does NOT properly manage appbundle.js so someone needs to take these changes and fix the script.

Discussion in PR https://github.com/hedyorg/hedy/pull/4060

**How to test**

1. Run the code coverage script as normal
2. See that you are able to inspect source of TypeScript files in `static/js` on code coverage webpage (see screenshot below)
3. See if there is now an `appbundle_orig.js` untracked in your repo. If so, then this script is messing with that incorrectly.

![Screenshot from 2023-03-18 00-52-45](https://user-images.githubusercontent.com/602877/226094111-f33dc74e-db69-41ac-8a16-121ad4ac7012.png)
